### DISCUSSION TITLE
Help, how should I sort and customize the name? At present, it seems that the document has not taken effect? The traceEvents type I set to M is not effective

### DISCUSSION THREAD

**Discussion by souldjl** [google/perfetto#956] [Q&A] [answered]
Help me, it's very good to rewrite such a good tool, but I encountered some problems while using this tool, roughly as follows
1. Sorting: How should I sort PID and TID? At present, it seems that the document has not taken effect? I set sort_idex according to the document, but it doesn't seem to take effect
2. Custom name: The traceEvents type I set to M did not take effect. Additionally, this number doesn't seem to have any rules and I want to remove it
3. MetaData: It seems that MetaData is not effective either.

Please find attached my screenshot below
![Image](https://github.com/user-attachments/assets/338aa469-2eaa-4335-8056-40454577c71f)
![Image](https://github.com/user-attachments/assets/ab384276-bd37-4877-a83b-6a683feadac0)
![Image](https://github.com/user-attachments/assets/55e6f72d-10f4-466c-98ec-aad619773f90)

Looking forward to someone replying to me，thanks

this is my json
[output.json](https://github.com/user-attachments/files/17987083/output.json)

**Comment by LalitMaganti** [comment, marked as answer]
> Sorting: How should I sort PID and TID? At present, it seems that the document has not taken effect? I set sort_idex according to the document, but it doesn't seem to take effect

This has been discussed many times before. If you search existing issues, you will see it at https://github.com/google/perfetto/issues/555, https://github.com/google/perfetto/issues/764 and https://github.com/google/perfetto/issues/378

The TDLR is that we do not support the process_sort_index of the JSON spec and there is no workaround in the JSON format. If you were to emit the native Perfetto format, we added support for custom sort orders for non-process and thread based data as described at https://perfetto.dev/docs/reference/synthetic-track-event#track-sorting-order

> Custom name: The traceEvents type I set to M did not take effect. Additionally, this number doesn't seem to have any rules and I want to remove it

All of those numbers correspond to the tid/pid of the fake threads/processes you are emitting in the JSON. There is currently no option to not show them but it would be possible for someone to write a patch for this. It's pretty easy I think to have an option to not show the information: contributions welcome.

Again if you were emitting the Perfetto native format, you don't need to emit fake threads/processes, you can have arbitrary grouping and the numbers will not be shown in that case.

> MetaData: It seems that MetaData is not effective either.

Once again, this is a part of the JSON format which is not ingested by Perfetto. There is an existing issue for this at https://github.com/google/perfetto/issues/595
